### PR TITLE
Create PublicAPI.Unshipped.txt

### PR DESCRIPTION
**Pull request by TanayParikh**
Fixes: https://github.com/dotnet/aspnetcore/pull/24258#issuecomment-663687515

Still getting `RS0026` (“Do not add multiple public overloads with optional parameters”) 

```
Extensions\HttpResponseWritingExtensions.cs(26,28): error RS0026: Symbol 'WriteAsync' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/master/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [C:\aspnetcore\src\Http\Http.Abstractions\src\Microsoft.AspNetCore.Http.Abstractions.csproj]
Extensions\HttpResponseWritingExtensions.cs(49,28): error RS0026: Symbol 'WriteAsync' violates the backcompat requirement: 'Do not add multiple overloads with optional parameters'. See 'https://github.com/dotnet/roslyn/blob/master/docs/Adding%20Optional%20Parameters%20in%20Public%20API.md' for details. [C:\aspnetcore\src\Http\Http.Abstractions\src\Microsoft.AspNetCore.Http.Abstractions.csproj]
```

### CODE REVIEW

**Comment by TanayParikh** [comment]
@dougbu is this the intended resolution to https://github.com/dotnet/aspnetcore/pull/24258#issuecomment-663685948?

**Comment by TanayParikh** [comment]
Hmm, CI not passing, going through it again.

Wasn't getting RS0016 locally with these changes.

**Comment by dougbu** [comment]
No, get rid of the src/Http files and you'll be done. My setup was an example for adding a brand new API baseline. Sorry if I didn't make that clear.

The updated _existing_ baseline files look good.

File: src/Mvc/Mvc.Abstractions/src/PublicAPI.Unshipped.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@ Microsoft.AspNetCore.Mvc.ActionConstraints.ActionConstraintContext.Candidates.ge
 Microsoft.AspNetCore.Mvc.ActionConstraints.ActionConstraintContext.Candidates.set -> void
 Microsoft.AspNetCore.Mvc.ActionConstraints.ActionConstraintContext.CurrentCandidate.get -> Microsoft.AspNetCore.Mvc.ActionConstraints.ActionSelectorCandidate
 Microsoft.AspNetCore.Mvc.ActionConstraints.ActionConstraintContext.CurrentCandidate.set -> void
-Microsoft.AspNetCore.Mvc.ActionConstraints.ActionConstraintContext.RouteContext.get -> Microsoft.AspNetCore.Routing.RouteContext!
+Microsoft.AspNetCore.Mvc.ActionConstraints.ActionConstraintContext.RouteContext.get -> RouteContext!
```
Comment (dougbu) [inline src/Mvc/Mvc.Abstractions/src/PublicAPI.Unshipped.txt:54]: On second thought, I'm not sure why these lines are changing. Suggest running through the update steps I described in email using `src/Mvc/startvs.cmd` instead. Something may be missing.

**Comment by dougbu** [comment]
> Wasn't getting RS0016 locally with these changes.

In more detail, this was my fault for not making the example clear. Using `src/Mvc/startvs.cmd` I'd expect a little more to change e.g. in Mvc.Core's API baselines.

**Comment by TanayParikh** [comment]
Retrying, thanks

**Comment by dougbu** [comment]
Thanks @TanayParikh /fyi the RS0026 errors in src/Http are expected and need to be suppressed. But, let's not mix adding new API baselines into this PR.

**Review by dougbu — APPROVED** [review]
These changes look on-target. @pranavkm you might want to refer to this PR in your API Review 😺

**Review by Pilchie — APPROVED** [review]

File: src/Mvc/Mvc.Core/src/ModelBinding/Metadata/DefaultBindingMetadataProvider.cs [hunk 2]
```diff
@@ -143,7 +143,8 @@ private static ConstructorInfo GetRecordTypeConstructor(Type type, ConstructorIn
             static bool IsRecordType(Type type)
             {
                 // Based on the state of the art as described in https://github.com/dotnet/roslyn/issues/45777
-                var cloneMethod = type.GetMethod("<>Clone", BindingFlags.Public | BindingFlags.Instance);
+                var cloneMethod = type.GetMethod("<Clone>$", BindingFlags.Public | BindingFlags.Instance) ??
+                    type.GetMethod("<>Clone", BindingFlags.Public | BindingFlags.Instance);
```
Comment (dougbu) [inline src/Mvc/Mvc.Core/src/ModelBinding/Metadata/DefaultBindingMetadataProvider.cs:147]: Accidentally commented on a commit:
> Are we using a slightly different version of Roslyn in 'master'❔
> And, how did validation pass for the merge PR from preview8 (https://github.com/dotnet/aspnetcore/pull/24258/checks)❔

Comment (TanayParikh) [reply src/Mvc/Mvc.Core/src/ModelBinding/Metadata/DefaultBindingMetadataProvider.cs:147]: cc/ @pranavkm

**Comment by dougbu** [comment]
Yippee 🥇

**Pull request merged**